src/containers/BookList.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import Book from '../components/Book';
+import CategoryFilter from '../components/CategoryFilter';
+import filterBooks from '../filters/booksFilter';
+
+const FILTEROPTIONS = ['All', 'Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
```

Comment:
**Required**: You have already created this array at some other place, why repeat it here?

You can import the book categories array that you already have in the `CategoryFilter` component and use it there, merging the 'All' option.

![](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/AllReturnsAllCategories.png)